src/mame/konami/spy.cpp

```diff
@@ -285,64 +299,21 @@ void spy_state::bankswitch_w(uint8_t data)
 	membank("bank1")->set_entry(bank);
 }
 
-void spy_state::spy_collision(  )
+void spy_state::pmc_run(  )
 {
 #define MAX_SPRITES 64
 #define DEF_NEAR_PLANE 0x6400
 #define NEAR_PLANE_ZOOM 0x0100
 #define FAR_PLANE_ZOOM 0x0000
 
-	int op1, x1, w1, z1, d1, y1, h1;
-	int op2, x2, w2, z2, d2, y2, h2;
-	int mode, i, loopend, nearplane;
-
-	mode = m_pmcram[0x1];
-	op1 = m_pmcram[0x2];
-	if (op1 == 1)
-	{
-		x1 = (m_pmcram[0x3] << 8) + m_pmcram[0x4];
-		w1 = (m_pmcram[0x5] << 8) + m_pmcram[0x6];
-		z1 = (m_pmcram[0x7] << 8) + m_pmcram[0x8];
-		d1 = (m_pmcram[0x9] << 8) + m_pmcram[0xa];
-		y1 = (m_pmcram[0xb] << 8) + m_pmcram[0xc];
-		h1 = (m_pmcram[0xd] << 8) + m_pmcram[0xe];
-
-		for (i = 16; i < 14 * MAX_SPRITES + 2; i += 14)
-		{
-			op2 = m_pmcram[i];
-			if (op2 || mode == 0x0c)
-			{
-				x2 = (m_pmcram[i + 0x1] << 8) + m_pmcram[i + 0x2];
-				w2 = (m_pmcram[i + 0x3] << 8) + m_pmcram[i + 0x4];
-				z2 = (m_pmcram[i + 0x5] << 8) + m_pmcram[i + 0x6];
-				d2 = (m_pmcram[i + 0x7] << 8) + m_pmcram[i + 0x8];
-				y2 = (m_pmcram[i + 0x9] << 8) + m_pmcram[i + 0xa];
-				h2 = (m_pmcram[i + 0xb] << 8) + m_pmcram[i + 0xc];
-/*
-    The mad scientist's laser truck has both a high sprite center and a small height value.
-    It has to be measured from the ground to detect correctly.
-*/
-				if (w2 == 0x58 && d2 == 0x04 && h2 == 0x10 && y2 == 0x30)
-					h2 = y2;
-
-				// what other sprites fall into:
-				if ((abs(x1 - x2) < w1 + w2) && (abs(z1 - z2) < d1 + d2) && (abs(y1 - y2) < h1 + h2))
-				{
-					m_pmcram[0xf] = 0;
-					m_pmcram[i + 0xd] = 0;
-				}
-				else
-					m_pmcram[i + 0xd] = 1;
-			}
-		}
-	}
-	else if (op1 > 1)
+	int a_pos, a_size, a_min, a_max, b_pos, b_size, b_min, b_max;
+	int count, i, j, tests_failed, loopend, nearplane, op2;
```

Comment:
Please scope these down to where they’re used.  In particular:
* You should be able to declare `i` and `j` in the loop control expressions since they’re only valid during the loops.
* `loopend` and `nearplane` should only exist in one branch of the `if`/`else`.
* Most of the other things here have a logical lifetime of a single loop iteration, and several of them should be `const`.